internal/status/status.go

```diff
@@ -0,0 +1,32 @@
+package status
+
+import (
+	"fmt"
+	"github.com/golang/protobuf/proto"
+	spb "google.golang.org/genproto/googleapis/rpc/status"
+	"google.golang.org/grpc/codes"
+)
+
+// StatusError is an alias of a status proto.  It implements error and Status,
+// and a nil StatusError should never be returned by this package.
+type StatusError spb.Status
+
+func (se *StatusError) Error() string {
+	p := (*spb.Status)(se)
+	return fmt.Sprintf("rpc error: code = %s desc = %s", codes.Code(p.GetCode()), p.GetMessage())
+}
+
+func (se *StatusError) GRPCStatus() *spb.Status {
```

Comment:
I'd imagine this change in public API may not be OK... Or it may since I'm moving this type to `internal`.

If I try to return a status.Status, I'll end up encountering an import cycle. @dfawley let me know what you think!

Reply:
ARGH!  Import cycles.  Would something like this work?

```
grpc/internal/status/statustype/statustype.go:

package statustype

type Status struct {
// move grpc/status/status.Status here
}

// Constructors/getters
```

```
grpc/status/status.go:

package status

type Status = statustype.Status
```

```
grpc/internal/status/status.go:

package status

func (se *StatusError) GRPCStatus() *statustype.Status {
  ...
}
```

Reply:
ping @JNProtzman

Reply:
Alright, so I've tried this again - I run into an import cycle with statustype, because the Err() method returns a `(*status.StatusError)(s.Proto())`. If I make Status satisfy the error interface, the tests fail because of a type assertion to `*status.StatusError`. 
Not sure of the right path forward!

Reply:
Hmmm, true.  `Status` needs to reference `StatusError` and vice-versa.  It's been awhile since I looked at this, but now I'm not sure why I was suggesting a third package at all.  Can `Status` be defined in `internal` as well (in the same package as `StatusError`), with the same type alias from `grpc/status` for `status.Status`?

Reply:
Ok, I got that working! Tests pass (locally at least) so I'm muuuch more comfortable with this change.

Reply:
I agree, this looks good, thanks!